engines/macventure/script.cpp

```diff
@@ -0,0 +1,1213 @@
+				opc1PTXT(state, frame);
+				break;
+			case 0xc2: //print newline
+				opc2PNEW(state, frame);
+				break;
+			case 0xc3: //print text+nl
+				opc3PTNE(state, frame);
+				break;
+			case 0xc4: //print nl+text+nl
+				opc4PNTN(state, frame);
+				break;
+			case 0xc5: //print number
+				opc5PNUM(state, frame);
+				break;
+			case 0xc6: //push 2
+				opc6P2(state, frame);
+				break;
+			case 0xc7: //play sound in background
+				opc7PLBG(state, frame);
+				break;
+			case 0xc8: //play sound and wait
+				opc8PLAW(state, frame);
+				break;
+			case 0xc9: //wait for sound to finish?
+				opc9WAIT(state, frame);
+				break;
+			case 0xca: //get current time
+				opcaTIME(state, frame);
+				break;
+			case 0xcb: //get current day
+				opcbDAY(state, frame);
+				break;
+			case 0xcc: //get children
+				opccCHLD(state, frame);
+				break;
+			case 0xcd: //get num children
+				opcdNCHLD(state, frame);
+				break;
+			case 0xce: //get engine version
+				opceVERS(state, frame);
+				break;
+			case 0xcf: //push scenario number
+				opcfPSCE(state, frame);
+				break;
+			case 0xd0: //push 1
+				opd0P1(state, frame);
+				break;
+			case 0xd1: //get object dimensions
+				opd1GOBD(state, frame);
+				break;
+			case 0xd2: //get overlap percent
+				opd2GOVP(state, frame);
+				break;
+			case 0xd3: //capture children
+				opd3CAPC(state, frame);
+				break;
+			case 0xd4: //release children
+				opd4RELC(state, frame);
+				break;
+			case 0xd5: //show speech dialog
+				opd5DLOG(state, frame);
+				return true;
+			case 0xd6: //activate command
+				opd6ACMD(state, frame);
+				break;
+			case 0xd7: //lose game
+				opd7LOSE(state, frame);
+				break;
+			case 0xd8: //win game
+				opd8WIN(state, frame);
+				break;
+			case 0xd9: //sleep
+				opd9SLEEP(state, frame);
+				return true;
+			case 0xda: //click to continue
+				opdaCLICK(state, frame);
+				return true;
+			case 0xdb: //run queue
+				opdbROBQ(state, frame);
+				break;
+			case 0xdc: //run sound queue
+				opdcRSQ(state, frame);
+				break;
+			case 0xdd: //run text queue
+				opddRTQ(state, frame);
+				break;
+			case 0xde: //update screen
+				opdeUPSC(state, frame);
+				break;
+			case 0xdf: //flash main window
+				opdfFMAI(state, frame);
+				return true;
+			case 0xe0: //cache graphic and object
+				ope0CHGR(state, frame);
+				break;
+			case 0xe1: //cache sound
+				ope1CHSO(state, frame);
+				break;
+			case 0xe2: //muldiv
+				ope2MDIV(state, frame);
+				break;
+			case 0xe3: //update object
+				ope3UPOB(state, frame);
+				break;
+			case 0xe4: //currently playing event?
+				ope4PLEV(state, frame);
+				break;
+			case 0xe5: //wait for event to finish
+				ope5WEV(state, frame);
+				break;
+			case 0xe6: //get fibonacci (joke)
+				ope6GFIB(state, frame);
+				break;
+			case 0xe7: //calc fibonacci
+				ope7CFIB(state, frame);
+				break;
+			default:
+				op00NOOP(op);
+			}
+		}
+	}
+	return false;
+}
+
+word ScriptEngine::neg16(word val) {
```

Comment:
It's actually int16, the format guide didn't specify it and I `typedef`'d it until I knew what it was. I started with uint16, but it was dead wrong. Anyways, I personaly prefer to have it `typedef`'d, but of course it can be changed.